lib/lltz_michelson/lltz_michelson.ml

```diff
@@ -70,191 +71,203 @@ and convert_constant (const: LLTZ.E.constant) : Michelson.Ast.t =
+  match prim with
+  | Amount -> amount
+  | Balance -> balance
+  | Chain_id -> chain_id
+  | Level -> level
+  | Now -> now
+  | Self opt -> self opt
+  | Self_address -> self_address
+  | Sender -> sender
+  | Source -> source
+  | Total_voting_power -> total_voting_power
+  | Empty_bigmap (ty1, ty2) -> empty_big_map (convert_type ty1) (convert_type ty2)
+  | Empty_map (ty1, ty2) -> empty_map (convert_type ty1) (convert_type ty2)
+  | Empty_set cty -> empty_set (convert_type cty)
+  | Nil ty -> nil (convert_type ty)
+  | None ty -> none (convert_type ty)
+  | Sapling_empty_state { memo } -> sapling_empty_state (Michelson.Ast.int memo)
+  | Unit -> unit
+  | Car -> car
+  | Cdr -> cdr
+  | Left (opt1, opt2, ty) -> left (convert_type ty) (* TODO: resolve tag options *)
+  | Right (opt1, opt2, ty) -> right (convert_type ty) (* TODO: resolve tag options *)
+  | Some -> some
+  | Eq -> eq
+  | Abs -> abs
+  | Neg -> neg
+  | Nat -> int
+  | Int -> int
+  | Bytes -> pack (* Assuming pack handles bytes conversion *)
+  | Is_nat -> is_nat
+  | Neq -> neq
+  | Le -> le
+  | Lt -> lt
+  | Ge -> ge
+  | Gt -> gt
+  | Not -> not
+  | Size -> size
+  | Address -> address
+  | Implicit_account -> implicit_account
+  | Contract (opt, ty) -> contract (convert_type ty) (* TODO: resolve tag option*)
+  | Pack -> pack
+  | Unpack ty -> unpack (convert_type ty)
+  | Hash_key -> hash_key
+  | Blake2b -> blake2b
+  | Sha256 -> sha256
+  | Sha512 -> sha512
+  | Keccak -> keccak
+  | Sha3 -> sha3
+  | Set_delegate -> set_delegate
+  | Read_ticket -> read_ticket
+  | Join_tickets -> join_tickets
+  | Pairing_check -> pairing_check
+  | Voting_power -> voting_power
+  | Getn n -> get_n n
+  | Cast ty -> cast (convert_type ty)
+  | Rename opt -> failwith (* TODO: Check why the instruction does not exist. *)
+  | Emit (opt, ty_opt) -> emit opt (Option.map convert_type ty_opt)
+  | Failwith -> failwith
+  | Never -> never
+  | Pair (opt1, opt2) -> pair (* TODO: resolve tag options*)
+  | Add -> add
+  | Mul -> mul
+  | Sub -> sub
+  | Sub_mutez -> sub
+  | Lsr -> lsr_
+  | Lsl -> lsl_
+  | Xor -> xor
+  | Ediv -> ediv
+  | And -> and_
+  | Or -> or_
+  | Cons -> cons
+  | Compare -> compare
+  | Concat1 -> concat
+  | Concat2 -> concat
+  | Get -> get
+  | Mem -> mem
+  | Exec -> exec
+  | Apply -> apply
+  | Sapling_verify_update -> sapling_verify_update
+  | Ticket -> ticket
+  | Ticket_deprecated -> ticket_deprecated
+  | Split_ticket -> split_ticket
+  | Updaten n -> update_n n
+  | View (name, ty) -> view name (convert_type ty)
+  | Slice -> slice
+  | Update -> update
+  | Get_and_update -> get_and_update
+  | Transfer_tokens -> transfer_tokens
+  | Check_signature -> check_signature
+  | Open_chest -> open_chest
+;;
 
 let rec compile : LLTZ.E.t -> Instruction.t =
   fun expr ->
-   let open Instruction in
-   Instruction.seq
-     [ 
-       match expr.desc with 
+  let open Instruction in
+  Instruction.seq
+    [ (match expr.desc with
```

Comment:
This pattern matching is way too long because of the rhs. Try and define functions for each case.